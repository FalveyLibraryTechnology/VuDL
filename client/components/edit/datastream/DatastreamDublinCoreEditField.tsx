import React from "react";
import FormControl from "@mui/material/FormControl";
import BlurSavingTextField from "../../shared/BlurSavingTextField";
import NativeSelect from "@mui/material/NativeSelect";
import TextField from "@mui/material/TextField";

interface DatastreamDublinCoreEditFieldProps {
    value: string;
    setValue: (value: string) => void;
    fieldType: string;
    legalValues?: Array<string>;
}

const DatastreamDublinCoreEditField = ({
    value,
    setValue,
    fieldType,
    legalValues = [],
}: DatastreamDublinCoreEditFieldProps): React.ReactElement => {
    const eventHelperCallback = (e) => {
        setValue(e.target.value);
    };
    switch (fieldType) {
        case "dropdown":
            if (!legalValues.includes(value)) {
                legalValues.push(value);
            }
            return (
                <FormControl fullWidth={true}>
                    <NativeSelect value={value} onChange={eventHelperCallback}>
                        {legalValues.map((current, index) => {
                            return (
                                <option key={`${current}_${index}`} value={current}>
                                    {current}
                                </option>
                            );
                        })}
                    </NativeSelect>
                </FormControl>
            );
        case "html":
            return (
                <textarea style={{ width: "100%" }} onBlur={eventHelperCallback}>
                    {value}
                </textarea>
            );
        case "locked":
            return (
                <FormControl fullWidth={true}>
                    <TextField type="text" value={value} disabled={true} />
                </FormControl>
            );
        case "text":
            return (
                <FormControl fullWidth={true}>
                    <BlurSavingTextField value={value} setValue={setValue} />
                </FormControl>
            );
        default:
            return <p>{`Unsupported type (${fieldType}): ${value}`}</p>;
    }
};

export default DatastreamDublinCoreEditField;
