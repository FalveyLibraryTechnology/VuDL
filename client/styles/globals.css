:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
