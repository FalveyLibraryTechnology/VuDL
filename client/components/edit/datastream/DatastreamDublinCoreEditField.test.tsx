import React from "react";
import { describe, afterEach, expect, it, jest } from "@jest/globals";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import DatastreamDublinCoreEditField from "./DatastreamDublinCoreEditField";

describe("DatastreamDublinCoreEditField", () => {
    let setValue: () => void;
    let fieldType: string;
    let legalValues: Array<string>;

    function getField(): React.ReactElement {
        const props = {
            value: "foo",
            setValue,
            fieldType,
            legalValues,
        };
        return <DatastreamDublinCoreEditField {...props} />;
    }

    beforeEach(() => {
        setValue = jest.fn();
        fieldType = "text";
        legalValues = [];
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it("renders a text field", () => {
        const wrapper = shallow(getField());

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders an html field", () => {
        fieldType = "html";
        const wrapper = shallow(getField());

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders an unknown field", () => {
        fieldType = "unknown";
        const wrapper = shallow(getField());

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders a locked field", () => {
        fieldType = "locked";
        const wrapper = shallow(getField());

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders a dropdown field", () => {
        fieldType = "dropdown";
        legalValues = ["foo", "bar", "baz"];
        const wrapper = shallow(getField());

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders a dropdown field with an unexpected value", () => {
        fieldType = "dropdown";
        const wrapper = shallow(getField());

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
