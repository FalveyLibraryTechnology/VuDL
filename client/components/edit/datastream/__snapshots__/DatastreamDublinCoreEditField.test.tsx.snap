// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatastreamDublinCoreEditField renders a dropdown field 1`] = `
<ForwardRef(FormControl)
  fullWidth={true}
>
  <ForwardRef(NativeSelect)
    onChange={[Function]}
    value="foo"
  >
    <option
      key="foo_0"
      value="foo"
    >
      foo
    </option>
    <option
      key="bar_1"
      value="bar"
    >
      bar
    </option>
    <option
      key="baz_2"
      value="baz"
    >
      baz
    </option>
  </ForwardRef(NativeSelect)>
</ForwardRef(FormControl)>
`;

exports[`DatastreamDublinCoreEditField renders a dropdown field with an unexpected value 1`] = `
<ForwardRef(FormControl)
  fullWidth={true}
>
  <ForwardRef(NativeSelect)
    onChange={[Function]}
    value="foo"
  >
    <option
      key="foo_0"
      value="foo"
    >
      foo
    </option>
  </ForwardRef(NativeSelect)>
</ForwardRef(FormControl)>
`;

exports[`DatastreamDublinCoreEditField renders a locked field 1`] = `
<ForwardRef(FormControl)
  fullWidth={true}
>
  <ForwardRef(TextField)
    disabled={true}
    type="text"
    value="foo"
  />
</ForwardRef(FormControl)>
`;

exports[`DatastreamDublinCoreEditField renders a text field 1`] = `
<ForwardRef(FormControl)
  fullWidth={true}
>
  <BlurSavingTextField
    setValue={[MockFunction]}
    value="foo"
  />
</ForwardRef(FormControl)>
`;

exports[`DatastreamDublinCoreEditField renders an html field 1`] = `
<Editor
  cloudChannel="6"
  init={
    Object {
      "plugins": "code link",
    }
  }
  initialValue="foo"
  onBlur={[Function]}
  tinymceScriptSrc="/tinymce/tinymce.min.js"
/>
`;

exports[`DatastreamDublinCoreEditField renders an unknown field 1`] = `
<p>
  Unsupported type (unknown): foo
</p>
`;
